File: aula02-pilhas-filas/python/demo_pilha.py
from pilha import Pilha


def mostrar(rotulo, pilha):
    valores = pilha.valores_do_topo_para_base()
    estado = " ".join(map(str, valores)) if valores else "[vazia]"
    print(f"{rotulo:<25} {estado}")


p = Pilha()
mostrar("0. pilha criada:", p)
p.empilhar(10)
mostrar("1. empilhar(10):", p)
p.empilhar(20)
mostrar("2. empilhar(20):", p)
print(f"3. consultar_topo():      {p.consultar_topo()} (a pilha nao muda)")
print(f"4. desempilhar():         removeu {p.desempilhar()}")
mostrar("5. estado final:", p)

File: aula03-conjuntos/revisao/c/demo_pilha.c
#include <stdio.h>
#include "pilha.h"

static void mostrar_estado(const char *rotulo, const Pilha *pilha) {
    printf("%-25s ", rotulo);
    pilha_exibir(pilha);
}

int main(void) {
    Pilha pilha;
    int valor;
    pilha_criar(&pilha);

    mostrar_estado("0. pilha criada:", &pilha);
    pilha_empilhar(&pilha, 10);
    mostrar_estado("1. empilhar(10):", &pilha);
    pilha_empilhar(&pilha, 20);
    mostrar_estado("2. empilhar(20):", &pilha);

    pilha_consultar_topo(&pilha, &valor);
    printf("3. consultar_topo():      %d (a pilha nao muda)\n", valor);

    pilha_desempilhar(&pilha, &valor);
    printf("4. desempilhar():         removeu %d\n", valor);
    mostrar_estado("5. estado final:", &pilha);
    return 0;
}
